### PR TITLE
Feature: clickable links on message bar

### LINKED ISSUE #693: Feedback Bar for samples with clickable link
We need a new type of Message Bar in graph explorer that allows requesting feedback to our customers for specific queries.
Overview: [Onboarding survey to Graph Explorer](https://microsoft-my.sharepoint-df.com/:w:/p/sanooran/ERs8P6XhlzRCoLSQpFfcqYQBxEy4saBV-r3RvjZHjJynxg?e=22B1ZD)

This is the first Pull request attempting at this, using the Tip bar: https://github.com/microsoftgraph/microsoft-graph-devx-content/pull/51

**Requirements for the new Feedback Message Bar**

1. Allow clickable URL formatting (with Markdown, HTML code, or a custom tag you decide). This is needed for opening the survey.
2. All the links will be opened by default in a new tab.
3. Show the feedback bar independently of the user login: it should be shown even with the demo tenant data.
4. The message bar should persist after the user clicks "Run query".
5. The message bar should dismiss only when:
  - The user changes a sample
  - The user clicks X to close the bar.
  - The user clicks the link to open the survey.
6. If the user clicked the link or dismissed the bar with the X, the Message bar *for that sample* should not appear anymore. This can be achieved by storing a LOCAL cookie with the id of the sample when the user clicks the link or dismiss the feedback bar:
  - If the cookie is present next time the user clicks the sample, the feedback bar will not be shown.
  - If the user clears their cache / cookies, then the feedback bar will be shown again, nothing related to the user data should be store server-side.

@thewahome @bettirosengugi
[AB#6062](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/6062)

### LINKED ISSUE #705: Graph links in tip messages to populate the url address bar
### Is your feature request related to a problem? Please describe.
A sample query may contain a tip message that describes what to do with the query. A common scenario is a query can have a **token** to display an item in the url that can be replaced eg `{message-id}` or `{task-id}`. A tip message is used here to tell the user what query to run in order to get the data to replace the **token** with.

The user then highlights the url in the tip message, copies it into the url editor and runs the query to get the data they can use to replace the **token**.

### Describe the solution you'd like
The ask here is that once a tip message is displayed, the url described in the message can be made clickable and once clicked, it should replace the url displayed on the url editor

### Describe alternatives you've considered
Leaving it as is and letting the user copy the link and paste it themselves
[AB#6125](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/6125)

### PR DESCRIPTION
**Pull request by thewahome**
## Overview
Makes links available in the samples clickable

Fixes #693 
Fixes #705 

### Demo

Has clickable link: 
image

### Notes

Some tip messages will not be shown when the user is not signed in

#### Scenario 1 - External link
* Click the `list Azure AD gallery apps` example from the sidebar
* A tip message with a clickable link should be displayed
* The click should open a new tab to the feedback survey

#### Scenario 2 - Graph API link
* Run the `get ediscovery case` example from the sidebar
* A tip message with a clickable link should be displayed
* The click should populate the query input area with the clicked url

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
   "scripts": {
     "start": "node scripts/start.js",
     "build": "node scripts/build.js && node versioned-build.js",
-    "test": "node scripts/test.js --no-watch --testPathIgnorePatterns=src/tests/accessibility",
+    "test": "node scripts/test.js --watch --testPathIgnorePatterns=src/tests/accessibility",
```
Comment (ElinorW) [inline package.json:84]: why did you change this part?
won't it look for the accessibility test, which is a separate job?

Comment (thewahome) [reply package.json:84]: the `--watch` flag makes the tests watch for changes as you work. So if you change a file, as long as you had ran `npm run test` you would see the results of the tests immediately without running it again. with the `--no-watch` you need to run the tests command again

The `--testPathIgnorePatterns` flag is a different parameter that helps the runner know which file to ignore

**Review by ElinorW — APPROVED** [review]

**Review by jobala — CHANGES_REQUESTED** [review]
Why is the filename `status.util.ts` and not `status.ts`?

File: src/app/utils/status.util.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      return;
+    }
+  });
+  return exists;
+}
+
+export function extractUrl(value: string): any {
+  const matches = value.match(/\bhttps?:\/\/\S+/gi);
+  if (matches) {
+    return matches;
```
Comment (jobala) [inline src/app/utils/status.util.ts:43]: You can achieve the same with

```js
return value.match(...)
```

File: src/tests/utils/url-to-html.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { linkExists, extractUrl, replaceLinks, convertArrayToObject } from "../../app/utils/status.util";
+
+describe('url should', () => {
```
Comment (jobala) [inline src/tests/utils/url-to-html.spec.ts:3]: It's a best practise to put the test suit name here, something like `URL TestSuite`

**Review by jobala — APPROVED** [review]
